Clean up Dashboard > System and Settings > Login & Registration section

mlocati opened the pull request:
Fix #8566

mlocati commented:
## Account Options

When `Login form` is set to `Ask for username & password`, and we disable `Username required`, users are asket if the want lo select `Ask for email & password` for `Login form`.
I added to this PR also the mirror of it: if we deselect `Username required`, users should also choose to login via email and not via username.

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![v8](https://user-images.githubusercontent.com/928116/82325119-fa54c080-99da-11ea-8b64-9126474d4314.png) | ![v9](https://user-images.githubusercontent.com/928116/82325132-02146500-99db-11ea-800b-fa069eceb398.png) |
| ![v8-confirm1](https://user-images.githubusercontent.com/928116/82325161-0ccefa00-99db-11ea-92d3-14c51ca5772e.png) | ![v9-confirm1](https://user-images.githubusercontent.com/928116/82325183-148e9e80-99db-11ea-9a4c-00080f299136.png) |
| n/a | ![v9-confirm2](https://user-images.githubusercontent.com/928116/82325218-207a6080-99db-11ea-8a08-181da7239f75.png) |

mlocati commented:
## Login Destination

The Page Selector widget has a few problems in v9:

1. it's not clickable
2. the URL tooltip is misplaced
3. there's no X button

## Account Options

When `Login form` is set to `Ask for username & password`, and we disable `Username required`, users are asket if the want lo select `Ask for email & password` for `Login form`.
I added to this PR also the mirror of it: if we deselect `Username required`, users should also choose to login via email and not via username.

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![v8](https://user-images.githubusercontent.com/928116/82326371-00e43780-99dd-11ea-9f20-c8eced05ed6f.png) | ![v9](https://user-images.githubusercontent.com/928116/82326378-05a8eb80-99dd-11ea-8028-b1c6b34f5a68.png) |

mlocati commented:
## Public Profiles

- I moved the `Fall Back To Gravatar` checkbox down, so that it's near to the Gravatar options.
- I added the descriptions of the Gravatar rating and image set (not everybody knows what `G` or `mp` mean)
- I added some missing Gravatar image sets (`retro`, `robohash`, `blank`)
- I replaced the `mm` Gravatar image set with `mp` (I think it's a typo, I can't find any reference to `mm` in the [Gravatar docs site](https://en.gravatar.com/site/implement/images/))

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![v8](https://user-images.githubusercontent.com/928116/82327310-7ef50e00-99de-11ea-822c-e4e90dce187f.png) | ![v9](https://user-images.githubusercontent.com/928116/82327325-82889500-99de-11ea-95ba-4a4a8d53c67a.png) |

biplobice commented:
How about merging the login destination page to login options?
Also, you may want to hide the registration related settings when it's turned off https://github.com/concrete5/concrete5/pull/8606.

mlocati commented:
## Authentication Types

1. in concrete5 v8, the rows of disabled authenticated types had a `error` CSS class, which was wrong (the correct value should have been `danger`), that's why they have a white background instead of red one. In v9 I fixed this (but we can for sure remove it and let the disabled rows have a white background)
1. The breadcrumb contains the authentication type being modified 
1. when editing an authentication type, we had a separate `Enable`/`Disable` button; that was a bit problematic IMHO because people (at least I) fill-in the authentication type options, then click the `Enable` button, but the authentication type options weren't saved. So I added a checkbox in the page instead of the `Enable`/`Disable` buttons.
1. the group selection of the authentication types were using a plain `<select>` to choose the group to be entered upon automatic registration. I replaced them with the `GroupSelector` widget
1. I replaced the `Show/Hide password` this way:  
   | concrete5 v8 | concrete5 v9 |
   |:---:|:---:|
   | ![00 v8 pwd 0](https://user-images.githubusercontent.com/928116/82342969-87574400-99f2-11ea-9375-644d1fe1207f.png) | ![00 v9 pwd 0](https://user-images.githubusercontent.com/928116/82342989-8cb48e80-99f2-11ea-9057-ab89764f9fd3.png) |
   | ![00 v8 pwd 1](https://user-images.githubusercontent.com/928116/82343008-9211d900-99f2-11ea-9dd1-3842de2e062d.png) | ![00 v9 pwd 1](https://user-images.githubusercontent.com/928116/82343021-950cc980-99f2-11ea-853f-4967db1d13e2.png) |

Here's the list of pages:

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![0 v8](https://user-images.githubusercontent.com/928116/82343101-b4a3f200-99f2-11ea-935a-67652d5400ea.png) | ![0 v9](https://user-images.githubusercontent.com/928116/82343107-b7064c00-99f2-11ea-885a-be0871368348.png) |
| ![1 v8 standard](https://user-images.githubusercontent.com/928116/82343115-ba99d300-99f2-11ea-92f2-b705f1246604.png) | ![1 v9 standard](https://user-images.githubusercontent.com/928116/82343122-bcfc2d00-99f2-11ea-972d-1e699f9a96c8.png) |
| ![2 v8 community](https://user-images.githubusercontent.com/928116/82343126-bec5f080-99f2-11ea-814c-490a6398362c.png) | ![2 v9 community](https://user-images.githubusercontent.com/928116/82343136-c1c0e100-99f2-11ea-92b1-6338c7dc9f05.png) |
| ![3 v8 facebook](https://user-images.githubusercontent.com/928116/82343150-c4233b00-99f2-11ea-9721-693ff66187c4.png) | ![3 v9 facebook](https://user-images.githubusercontent.com/928116/82343152-c5ecfe80-99f2-11ea-9ee5-de6ec372d43a.png) |
| ![4 v8 twitter](https://user-images.githubusercontent.com/928116/82343162-c7b6c200-99f2-11ea-853e-6b9233109ebe.png) | ![4 v9 twitter](https://user-images.githubusercontent.com/928116/82343166-c8e7ef00-99f2-11ea-9a3a-bc0637ee0ec5.png) |
| ![5 v8 google](https://user-images.githubusercontent.com/928116/82343170-cb4a4900-99f2-11ea-82c1-972514d250cd.png) | ![5 v9 google](https://user-images.githubusercontent.com/928116/82343176-cc7b7600-99f2-11ea-9ab2-daf71ff032ab.png) |
| ![6 v8 external_concrete5](https://user-images.githubusercontent.com/928116/82343182-ceddd000-99f2-11ea-852b-9a2e7e4a898e.png) | ![6 v9 external_concrete5](https://user-images.githubusercontent.com/928116/82343185-d00efd00-99f2-11ea-88e6-037050a21cf5.png) |

mlocati commented:
## Global Password Reset

1. I added a feedback when users have to enter the `RESET` text to confirm the operation (like we have in the `Automated Logout` page)
1. users were aware that this operation is only for the super user only upon form submission, now it's clear immediately when they open the window

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![v8](https://user-images.githubusercontent.com/928116/82348465-36971980-99f9-11ea-8c0b-a633ddef8b8f.png) | ![v9](https://user-images.githubusercontent.com/928116/82348494-40208180-99f9-11ea-80ce-0c5a7c8f99ba.gif) |

mlocati commented:
## Notification Settings

Here I had to touch a bit more files, since I had a few "accessing undefined array index" warnings. In these files we also use the deprecated tools, and it'd be fine to use routes instead (not included in this PR).

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![v8](https://user-images.githubusercontent.com/928116/82351846-6b0cd480-99fd-11ea-908a-655a4f52a7b8.png) | ![v9](https://user-images.githubusercontent.com/928116/82351854-6d6f2e80-99fd-11ea-944e-536496695aea.png) |

mlocati commented:
## User Deactivation Settings

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![v8 0](https://user-images.githubusercontent.com/928116/82352760-ba9fd000-99fe-11ea-8b2e-43d5f0711791.png) | ![v9 0](https://user-images.githubusercontent.com/928116/82352769-bd9ac080-99fe-11ea-8e09-868bfbe639ae.png) |
| ![v8 1](https://user-images.githubusercontent.com/928116/82352773-bf648400-99fe-11ea-837e-20f1997bb478.png) | ![v9 1](https://user-images.githubusercontent.com/928116/82352778-c1c6de00-99fe-11ea-8131-f44b58bc65e5.png) |

mlocati commented:
## Automated Logout

I added a confirmation dialog box before proceeding with the global session invalidation.

| concrete5 v8 |  concrete5 v9 |
|:---:|:---:|
| ![v8](https://user-images.githubusercontent.com/928116/82356424-d5c10e80-9a03-11ea-9b68-dc91b3feafe7.png) | ![v9](https://user-images.githubusercontent.com/928116/82356438-db1e5900-9a03-11ea-983b-89286ccf28fa.gif) |

mlocati commented:
## Password Requirements

Some improvements here:

- switch to Vue to handle the custom regular expressions in a much easier and cleaner way
- check the token upon saving the form
- restore the previously inserted regular expressions in case the form validation fails, for example if a regular expression is wrong
- improve readability in smaller viewports

| concrete5 v8 | concrete5 v9 |
|:---:|:---:|
| ![v8](https://user-images.githubusercontent.com/928116/82360908-5d118080-9a0a-11ea-9b10-4e2eb35679b7.png) | ![v9](https://user-images.githubusercontent.com/928116/82360912-5edb4400-9a0a-11ea-87d8-7bc9eb5907b9.png) |

mlocati commented:
This PR is ready for review.

mlocati commented:
> How about merging the `login destination page` settings page to `login options` settings page ?
> Also, you may want to hide the registration related settings when it's turned off #8606.

@biplobice Yep, that'd be nice. Let's do that once this PR will be merged.

aembler commented on concrete/single_pages/dashboard/system/registration/deactivation.php:
```diff
@@ -1,88 +1,78 @@
+    <div class="form-group">
+        <?= $form->label('inactiveMessage', t('Inactive User Error Message')) ?>
+        <?= $form->textarea('inactiveMessage', $inactiveMessage, ['rows' => 4, 'cols' => 10]) ?>
+        <small class="text-mited"><?= t('This message will be shown to inactive users when they attempt to login.') ?></small>
```
We should fix this post-merge, I t think this is the wrong class.

mlocati replied:
See https://github.com/concrete5/concrete5/pull/8693
